tests/unit_tests/restruct_poc/test_math.cpp

```diff
@@ -0,0 +1,369 @@
+#include <gtest/gtest.h>
+
+#include "src/restruct_poc/math/quaternion_operations.hpp"
+#include "src/restruct_poc/math/vector_operations.hpp"
+#include "tests/unit_tests/gen_alpha_poc/test_utilities.h"
+
+namespace openturbine::restruct_poc::tests {
+
+TEST(QuaternionTest, ConvertQuaternionToRotationMatrix_1) {
+    // 90 degree rotation about the x axis
+    auto q = Kokkos::View<double[4]>("q");
+    Kokkos::parallel_for(
+        1,
+        KOKKOS_LAMBDA(const int) {
+            q(0) = 0.707107;
+            q(1) = 0.707107;
+            q(2) = 0.;
+            q(3) = 0.;
+        }
```

Comment:
We can't declare KOKKOS_LAMBDAs like this in test code when building for device.  The easiest way to do this is probably
```
auto q_host = Kokkos::create_mirror(q);
q_host(0) = 0.707107;
q_host(1) = 0.707107;
q_host(2) = 0.;
q_host(3) = 0.;
Kokkos::deep_copy(q, q_host);
```
Looks like you'll need to do this in most of the tests below